clang/lib/Driver/Driver.cpp

```diff
@@ -1846,12 +1846,18 @@ void Driver::PrintHelp(bool ShowHidden) const {
 }
 
 llvm::Triple Driver::MakeSYCLDeviceTriple(StringRef TargetArch) const {
-  llvm::Triple TT;
-  TT.setArchName(TargetArch);
-  TT.setVendor(llvm::Triple::UnknownVendor);
-  TT.setOS(llvm::Triple::UnknownOS);
-  TT.setEnvironment(llvm::Triple::SYCLDevice);
-  return TT;
+  SmallVector<StringRef, 5> SYCLAlias = {"spir", "spir64", "spir64_fpga",
+                                         "spir64_x86_64", "spir64_gen"};
+  if (std::find(SYCLAlias.begin(), SYCLAlias.end(), TargetArch) !=
+      SYCLAlias.end()) {
+    llvm::Triple TT;
+    TT.setArchName(TargetArch);
+    TT.setVendor(llvm::Triple::UnknownVendor);
+    TT.setOS(llvm::Triple::UnknownOS);
+    TT.setEnvironment(llvm::Triple::SYCLDevice);
+    return TT;
+  }
+  return llvm::Triple(TargetArch);
```

Comment:
(Non-blocking comment) So is this the path that NTPTX/AMDGCN will take? Could/should we extend the aliasing logic for the non-SPIR targets?
I realize that in general, after `sycldevice` removal the flow will already be quite convenient for users of these targets, e.g. `-fsycl-targets=nvptx64-nvidia-nvcl`. But maybe a documented default of Vendor/OS would make CLI even better.

Reply:
Yes, all processing of `-fsycl-targets` will go through this path.  We could potentially expand to accommodate any other short aliases and change them to what we want pretty easily.

Reply:
Could you please file a GH issue or add a `TODO` for that?

Reply:
issue https://github.com/intel/llvm/issues/4010 created